### PR TITLE
Normalise automattic.com and wordpress.com URLs

### PR DESCRIPTION
**Pull request by sgomes**
#### Proposed Changes

Redirects are expensive, as they often involve an extra full roundtrip and extra server processing time, easily adding hundreds of milliseconds to TTFB and all subsequent metrics. This can have a measurable impact on Core Web Vitals scores for the domains in which the redirects are happening.

This PR tries to normalise all links to `automattic.com` and `wordpress.com` so that users don't have to go through any unnecessary redirects. This means:

- Using `https:` instead of `http:`
- On `automattic.com`, always using trailing slashes
- On `wordpress.com`, using trailing slashes for WP URLs, while using no trailing slashes for Calypso URLs 😕

#### Testing Instructions

I don't expect any breakage from this PR, so ensuring that the unit tests pass should be sufficient.

Some light smoke testing could be a good complement to that, as could trying out any URLs that you're not sure as to whether they land on WP or Calypso as a backend.

#### Pre-merge Checklist

- (N/A) [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- (N/A) Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- (N/A) Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- (N/A) Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

File: client/me/profile-links/index.jsx [hunk 1]
```diff
@@ -123,7 +123,7 @@ class ProfileLinks extends Component {
 				{ times( 2, ( index ) => (
 					<ProfileLink
 						title="Loading Profile Links"
-						url="http://wordpress.com"
+						url="https://wordpress.com/"
```
Comment (tyxla) [inline client/me/profile-links/index.jsx:126]: I just wanted to note that this change applies only to logged-out users. For logged-in users, we'll be redirected to our primary site anyway, so a redirect will always be needed.

**Review by tyxla — APPROVED** [review]
Thanks for that, @sgomes 🙌 

I've looked through them one by one, and they all make sense. 

I didn't realize we have so many `http://` links left 😅

**Review by noahtallen — COMMENTED** [review]
I wonder if there are any lint rules we could add to prevent this from happening again... though with the different backslash approaches, that might be tricky. 

Is Calypso really doing a full redirect away from trailing "/", or is it just a client-side URL adjustment?

**Comment by sgomes** [comment]
> I wonder if there are any lint rules we could add to prevent this from happening again... though with the different backslash approaches, that might be tricky.
> 
> Is Calypso really doing a full redirect away from trailing "/", or is it just a client-side URL adjustment?

@noahtallen You're right, it appears to be a client-side adjustment in all the routes I've tried! That said, it's probably best to keep those changes in if the canonical URL doesn't have a slash; that should help prevent actual redirects if the infrastructure ever changes.

**Comment by sgomes** [comment]
Thank you for the reviews, @tyxla and @noahtallen! 👍

File: client/me/profile-links/index.jsx (hunk 1 above)
Comment (sgomes) [reply client/me/profile-links/index.jsx:126]: These should be an internal redirect in modern browsers, so it's not quite as bad. But still, there's no reason to keep them around like that 🙂

**Pull request merged**